### PR TITLE
Optimize file path builder and have separate handler for streaming file

### PR DESCRIPTION
**Pull request by sarankk**
Co-authored-by: Francisco Guerrero

### CODE REVIEW

File: common/build.gradle [hunk 1]
```diff
@@ -22,6 +22,7 @@ test {
 }
 
 dependencies {
+    compile "io.vertx:vertx-web-api-contract:$vertxVersion"
```
Comment (frankgh) [inline common/build.gradle]: do we need this dependency?

File: src/main/java/org/apache/cassandra/sidecar/MainModule.java [hunk 2]
```diff
@@ -144,6 +148,22 @@ public Router vertxRouter(Vertx vertx, LoggerHandler loggerHandler, ErrorHandler
         // Docs index.html page
         StaticHandler docs = StaticHandler.create("docs");
         router.route().path("/docs/*").handler(docs);
+
+        // add custom routers
+        final String apiVersion = "/api/v1";
```
Comment (frankgh) [inline src/main/java/org/apache/cassandra/sidecar/MainModule.java]: do we have a constant for this?

File: common/src/main/java/org/apache/cassandra/sidecar/common/data/CassandraQualifiedName.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
+package org.apache.cassandra.sidecar.common.data;
+
+import org.apache.cassandra.sidecar.common.utils.ValidationUtils;
+
+/**
+ * Contains the keyspace and table name in Cassandra
+ */
+public class CassandraQualifiedName
```
Comment (yifan-c) [inline common/src/main/java/org/apache/cassandra/sidecar/common/data/CassandraQualifiedName.java]: _weak_ nit: `QualifiedTableName` seems more fit.

File: common/src/main/java/org/apache/cassandra/sidecar/common/utils/ValidationUtils.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+{
+    private static final Set<String> FORBIDDEN_DIRS = new HashSet<>(Arrays.asList("system_schema",
+                                                                                  "system_traces",
+                                                                                  "system_distributed",
+                                                                                  "system",
+                                                                                  "system_auth",
+                                                                                  "system_views",
+                                                                                  "system_virtual_schema"));
+    private static final Pattern PATTERN_WORD_CHARS = Pattern.compile("[a-zA-Z0-9_-]+");
+    private static final String REGEX_COMPONENT = "[a-zA-Z0-9_-]+(.db|.cql|.json|.crc32|TOC.txt)";
```
Comment (yifan-c) [inline common/src/main/java/org/apache/cassandra/sidecar/common/utils/ValidationUtils.java]: nit: `[a-zA-Z0-9_-]+` has been used all 3 patterns. Maybe declare a string constant.

File: src/main/java/org/apache/cassandra/sidecar/MainModule.java [hunk 5]
```diff
@@ -144,6 +148,22 @@ public Router vertxRouter(Vertx vertx, LoggerHandler loggerHandler, ErrorHandler
...
+        final String componentRoute = "/keyspace/:keyspace/table/:table/snapshots/:snapshot/component/:component";
+        final String defaultStreamRoute = apiVersion + componentRoute;
```
Comment (yifan-c) [inline src/main/java/org/apache/cassandra/sidecar/MainModule.java]: so... the change drops "stream" in the endpoint?

**Review by yifan-c — COMMENTED** [review]
Look good overall. 
The nits are weak. I am fine w/o addressing.

The question about the api change is concerning.

**Review by yifan-c — APPROVED** [review]

File: common/src/main/java/org/apache/cassandra/sidecar/common/SidecarProps.java [hunk 6]
```diff
@@ -0,0 +1,10 @@
+package org.apache.cassandra.sidecar.common;
+
+/**
+ * Properties maintained in common for all sub projects.
+ */
+public class SidecarProps
```
Comment (frankgh) [inline common/src/main/java/org/apache/cassandra/sidecar/common/SidecarProps.java]: do we need to introduce this class? Can the property be moved somewhere else?

File: src/main/java/org/apache/cassandra/sidecar/routes/StreamSSTableComponentHandler.java [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+        streamFilesForHost(host, context);
+    }
+
+    public void handlePerInstanceRequests(RoutingContext context)
+    {
+        final String instanceIdParam = context.request().getParam("InstanceId");
+        if (instanceIdParam == null)
+        {
+            context.fail(new HttpException(HttpResponseStatus.BAD_REQUEST.code(),
+                                           "Instance id param can't be missing"));
```
Comment (frankgh) [inline src/main/java/org/apache/cassandra/sidecar/routes/StreamSSTableComponentHandler.java]:
```suggestion
                                           "InstanceId path parameter must be provided"));
```

File: common/src/main/java/org/apache/cassandra/sidecar/common/data/CassandraQualifiedName.java (hunk 3 above)
Comment (frankgh) [reply common/src/main/java/org/apache/cassandra/sidecar/common/data/CassandraQualifiedName.java]: :( I liked `CassandraQualifiedName`, but either is fine I guess

File: src/main/java/org/apache/cassandra/sidecar/MainModule.java (hunk 5 above)
Comment (frankgh) [reply src/main/java/org/apache/cassandra/sidecar/MainModule.java]: I think this reverts a breaking change that introduced the stream, and we wanted to fix this in order not to break the API in minor versions

File: common/src/main/java/org/apache/cassandra/sidecar/common/SidecarProps.java (hunk 6 above)
Comment (sarankk) [reply common/src/main/java/org/apache/cassandra/sidecar/common/SidecarProps.java]: Moved that property to FileStreamer

File: common/build.gradle (hunk 1 above)
Comment (sarankk) [reply common/build.gradle]: We need either that or `io.vertx:vertx-web`, changed it to `io.vertx:vertx-web` to be consistent with main build.gradle

File: src/main/java/org/apache/cassandra/sidecar/MainModule.java (hunk 2 above)
Comment (sarankk) [reply src/main/java/org/apache/cassandra/sidecar/MainModule.java]: Added one now

**Review by frankgh — APPROVED** [review]

**Pull request closed without merging**